streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -3739,18 +3718,67 @@ object ZStream extends ZStreamPlatformSpecificConstructors {
   private[zio] object Pull {
     def emit[A](a: A): IO[Nothing, Chunk[A]]                               = UIO(Chunk.single(a))
     def emit[A](as: Chunk[A]): IO[Nothing, Chunk[A]]                       = UIO(as)
-    def fromDequeue[E, A](d: Dequeue[Take[E, A]]): IO[Option[E], Chunk[A]] = d.take.flatMap(IO.done(_))
-    def fromTake[E, A](t: Take[E, A]): IO[Option[E], Chunk[A]]             = IO.done(t)
+    def fromDequeue[E, A](d: Dequeue[Take[E, A]]): IO[Option[E], Chunk[A]] = d.take.flatMap(_.done)
     def fail[E](e: E): IO[Option[E], Nothing]                              = IO.fail(Some(e))
     def halt[E](c: Cause[E]): IO[Option[E], Nothing]                       = IO.halt(c).mapError(Some(_))
     def empty[A]: IO[Nothing, Chunk[A]]                                    = UIO(Chunk.empty)
     val end: IO[Option[Nothing], Nothing]                                  = IO.fail(None)
   }
 
-  type Take[+E, +A] = Exit[Option[E], Chunk[A]]
+  case class Take[+E, +A](exit: Exit[Option[E], Chunk[A]]) extends AnyVal {
```

Comment:
Let's move this out to a separate file, and add a `@deprecated` type alias that points to it.

Reply:
Done